### ISSUE TITLE
vk.xml's `define` category no longer restricted to actual #defines

### ISSUE THREAD

**Issue by NicolBolas** [KhronosGroup/Vulkan-Docs#679]
Vulkan 1.1 added the following to vk.xml:

    <type category="define">struct <name>ANativeWindow</name>;</type>

This is not a `#define`, yet it uses the `define` category. It ought to be a `typedef`, but there's probably some C language reason why this forward declaration can't be a typedef. I imagine `define` was used because it worked, because the C and C++ generators just dump whatever is in a `define` category without processing it.

If the idea is that `define` is just "random C/C++ stuff", then that should probably be well-stated in the documentation. But at present, all it says is "`define` (macro `#define` directives)".

**Comment by oddhack** [comment]
I agree it's not a proper use, though figuring out how to accommodate incomplete struct types will take some tweaking. It also results in spurious ref pages being generated.
